src/modularPlantManager/ModularPlantManager.ts

```diff
@@ -354,65 +345,31 @@ export class ModularPlantManager extends (EventEmitter as new() => ModularPlantM
 				};
 
 			// Initializing dataAssemblyName, dataAssemblyInterfaceClass
-			let dataAssemblyName ='', dataAssemblyInterfaceClass='';
-
-			// get dataAssemblyName from PEAModel
-			dataAssembly.getName(((response,name)=>{
-				// assign dataAssemblyName
-				dataAssemblyName = name;
-			}));
-
-			//get metamodelref from PEAModel/Pimad
-			dataAssembly.getMetaModelRef(((response,metaModelRef)=>{
-				// assign InterfaceClass/MetaModelRef
-				dataAssemblyInterfaceClass = metaModelRef;
-			}));
-
-			dataAssembly.getAllDataItems(((response1, dataItems) =>
-					dataItems.map(dataItem=>{
-						// Initializing variables, which will be assigned later
-						let namespaceIndex ='', nodeId='', dataType='', value: undefined | string;
-						// get value from PiMAd (only for TagName and TagDescription)
-						// get dataType from PiMAd
-						dataItem.getDataType((response, mDataType)=>
-							dataType= mDataType);
-						// get cIData from PiMAd
-						dataItem.getCommunicationInterfaceData((response2, cIData) => {
-							// check if dataItem has an cIData
-							if(cIData){
-								// get NodeId object from PiMAd
-								(cIData as OPCUANodeCommunication).getNodeId((response, nodeID) => {
-									// get NodeId from PiMAd and assign it to variable
-									nodeID.getNodeIdIdentifier((response, identifier) =>
-										nodeId = identifier);
-									// get namespaceIndex from PiMAd and assign it to variable
-									nodeID.getNamespaceIndex((response, mNamespace) =>{
-										namespaceIndex = mNamespace as unknown as string;
-										/**
-										 * use this for testing with test-pea with your own namespaceIndex
-										 * namespaceIndex='urn:DESKTOP-6QLO5BB:NodeOPCUA-Server';
-										 */
-									});
-								});
-							}else {
-								// it's a static DataItem
-								dataItem.getValue((response, mValue) => value = mValue);
-							}
-						});
-
-						const opcUaNodeOptions: OpcUaNodeOptions = {
-							nodeId: nodeId,
-							namespaceIndex: namespaceIndex,
-							dataType: dataType,
-							value: value
-						};
-						// get Name of DataItem from PiMAd
-						dataItem.getName((response2, name) =>
-						{
-							baseDataAssemblyOptions[name as string] = opcUaNodeOptions;
-						});
-					})
-			));
+			const dataAssemblyName =dataAssembly.name;
+			const dataAssemblyInterfaceClass= dataAssembly.metaModelRef;
+			const dataItems = dataAssembly.dataItems;
+			dataItems.map(dataItem=>{
+				// Initializing variables, which will be assigned later
+				let namespaceIndex ='', nodeId='', value: undefined | string;
+				const dataType=dataItem.dataType;
+				const cIData = dataItem.cIData;
+				if(cIData){
+					nodeId= cIData.nodeId.identifier;
+					namespaceIndex = cIData.nodeId.namespaceIndex;
+					//namespaceIndex='urn:DESKTOP-QNK1RES:NodeOPCUA-Server';
+				} else {
+					value = dataItem.value;
+				}
+
+				const opcUaNodeOptions: OpcUaNodeOptions = {
```

Comment:
OpcUaNodeOptions is may a bit tricky to understand in case there is no cI-Data and just the value
What would you do if there is another communication type?

Reply:
In general Tag name and Description are static MTP modeled, there will never be cI data included

Reply:
You're right. So for static MTP modeled, we don't need to create OPCUADataitems. In that case we still need to store the static information in the DataAssemblyController object.

Reply:
@julorenz can you still merge this branch. I will tackle this problem in a new branch, after merging this branch and features/services branch.